network/p2p/middleware/middleware.go

```diff
@@ -693,7 +667,7 @@ func (m *Middleware) processUnicastStreamMessage(remotePeer peer.ID, msg *messag
 
 	// if message channel is not public perform authorized sender validation
 	if !channels.IsPublicChannel(channel) {
-		_, err := m.authorizedSenderValidator.Validate(remotePeer, decodedMsgPayload, channel, message.ProtocolUnicast)
+		messageType, err := m.authorizedSenderValidator.Validate(remotePeer, msg.Payload[0], channel, message.ProtocolUnicast)
```

Comment:
same comment about not accessing the `Payload` slice directly

Reply:
https://github.com/onflow/flow-go/pull/3831/commits/4616da0d6e5b39e55b5b1b51efa8c50a7ff36f39